src/src/com/microsoft/aad/adal/StorageHelper.java

```diff
@@ -530,6 +500,7 @@ private synchronized KeyPair getKeyPairFromAndroidKeyStore() throws GeneralSecur
     
     @TargetApi(18)
     private Object getKeyPairGeneratorSpec(X500Principal subj, Date start, Date end) {
+        // TODO: use real Spec.Builder instead of reflective methods
```

Comment:
I filed an issue on this. https://github.com/AzureAD/azure-activedirectory-library-for-android/issues/533